web/views.py

```diff
@@ -200,18 +208,114 @@ def form_valid(self, form):
+
+        if form.is_valid():
+            for field in self.form_class().fields:
+                request.session[field] = form.data.get(field, "")
+
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
+
     def get_queryset(self):
-        # TODO if we need to filter by study demographics vs user demographics
-        # or by if they've taken the study before this is the spot
-        return super().get_queryset().filter(state="active", public=True).order_by("?")
+        session = self.request.session
+        user = self.request.user
+        page = self.request.GET.get("page", 1)
+
+        qs = super().get_queryset().filter(state="active", public=True)
+
+        # values from session
+        search_value = session.get("search", "")
+        child_value = session.get("child", "")
+        show_experiments_already_done_value = (
+            session.get("show_experiments_already_done", "") == "on"
+        )
+
+        if search_value:
+            qs = qs.filter(name__icontains=search_value)
+
+        # convert to list as it's no longer being treated as a queryset
+        studies = list(qs)
+
+        if child_value:
+            if user.is_anonymous:
+                # when user is anonymous, child value is an age range.
+                age_range = [int(c) for c in child_value.split(",")]
+                studies = [
+                    s for s in studies if age_range_eligibility_for_study(age_range, s)
+                ]
+            else:
+                # when user is authenticated, child value is a child pk
+                child = Child.objects.get(pk=child_value, user=user)
+
+                if not show_experiments_already_done_value:
+                    studies = self.completed_consent_frame(studies, child)
+
+                studies = [
+                    s for s in studies if get_child_eligibility_for_study(child, s)
+                ]
+
+        studies.sort(key=self.sort_fn())
+
+        self.child_eligibility(studies)
+
+        return self.paginated_queryset(studies, page)
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["user"] = self.request.user
+        return kwargs
+
+    def get_initial(self):
+        kwargs = super().get_initial()
+
+        # When field values are retrieved from session to populate form, they are removed from session
+        for field in self.form_class().fields:
+            if field in self.request.session:
+                kwargs[field] = self.request.session.pop(field)
+
+        return kwargs
+
+    def get_success_url(self):
+        return reverse("web:studies-list")
+
+    def completed_consent_frame(self, studies, child):
+        return [
+            r.study
+            for r in Response.objects.filter(
+                study__in=studies, child=child, completed_consent_frame=True,
+            ).distinct("study_id")
+        ]
+
+    def child_eligibility(self, studies):
```

Comment:
very minor nitpick - can we call this set_eligible_children_per_study or some similar (better) name that describes the side effect here?